api/api.go

```diff
@@ -258,7 +260,9 @@ func New(
 		}
 	}
 
-	c.authenticate(ctx, username, password, hostname)
+	if c.authType == 1 {
```

Comment:
Can you add two constants authTypeBasic and authTypeSessionBased, and use the same wherever required instead of using values 0 and 1